config/fcos/v1_6_exp/translate.go

```diff
@@ -239,9 +248,21 @@ func (c Config) processBootDevice(config *types.Config, ts *translate.Translatio
 
 	// encrypted root partition
 	if wantLuks {
-		luksDevice := "/dev/disk/by-partlabel/root"
-		if wantMirror {
+		var luksDevice string
+		var device_s390x string
+		switch {
+		case wantMBR:
```

Comment:
So what is the difference between the work done in the `wantMBR` case and the `wantDasd` case? to my eye they are doing the same thing?

Reply:
I've explained above...

Reply:
For more readability I made the changes as follows.
```
	case *layout == "s390x-eckd":
		wantS390x = true
	case *layout == "s390x-zfcp":
		wantS390x = true
```
```
// encrypted root partition
	if wantLuks {
		var luksDevice string
		var device_s390x string
		switch {
		case wantS390x:
			//Luks Device for dasd and zFCP-scsi
			device_s390x = *c.BootDevice.Luks.Device
			luksDevice = device_s390x + "2"
```